gcsa/_services/authentication.py

```diff
@@ -107,7 +111,22 @@ def _get_credentials(
             else:
                 credentials_path = os.path.join(credentials_dir, credentials_file)
                 flow = InstalledAppFlow.from_client_secrets_file(credentials_path, scopes)
-                credentials = flow.run_local_server(host=host, port=port, bind_addr=bind_addr)
+                try:
+                    credentials = flow.run_local_server(
+                        host=host, port=port, bind_addr=bind_addr, timeout_seconds=120,
+                        open_browser=open_browser
+                    )
+                except webbrowser.Error:
+                    # System has no default browser configured, retry without opening browser
+                    return AuthenticatedService._get_credentials(
+                        token_path, credentials_dir, credentials_file, scopes, save_token, host, port,
+                        bind_addr, open_browser=False
+                    )
+                except WSGITimeout:
```

Comment:
I decided not to incorporate timeout handling for `flow.run_local_server` a while back (#54) and just described the way to do it for somebody who needs it [in the docs](https://google-calendar-simple-api.readthedocs.io/en/latest/authentication.html#browser-authentication-timeout). It allows for a more fine-tuned approach (e.x. setting a timeout) and doesn't require the newest version of the `google-auth-oauthlib`.

Reply:
Interesting trick to use a separate process to provide a timeout. I might argue that it's a bit of a hack 😄 

I'm already running multiprocess async, and there's no technical blocker why I couldn't use this pattern. My 2c is that timeouts ought to be supported first-class in the library.

Would it be a problem to require the most recent `google-auth-oauthlib`, or are you concerned about cross-compatibility with other libraries?